network/src/network.rs

```diff
@@ -282,26 +284,32 @@ impl NetworkState {
         self.local_private_key().to_peer_id().to_base58()
     }
 
-    pub(crate) fn listened_addresses(&self, count: usize) -> Vec<(Multiaddr, u8)> {
-        self.listened_addresses
+    pub(crate) fn listened_addrs(&self, count: usize) -> Vec<(Multiaddr, u8)> {
+        self.listened_addrs
             .read()
             .iter()
             .take(count)
             .map(|(addr, score)| (addr.to_owned(), *score))
             .collect()
     }
 
+    pub(crate) fn score_listened_addr(&self, addr: Multiaddr, inc_score: u8) {
+        let mut listened_addrs = self.listened_addrs.write();
+        let addr_score = listened_addrs.entry(addr).or_default();
+        *addr_score = addr_score.saturating_add(inc_score);
+    }
+
     pub(crate) fn connection_status(&self) -> ConnectionStatus {
         self.peer_registry.read().connection_status()
     }
 
     pub fn external_urls(&self, max_urls: usize) -> Vec<(String, u8)> {
-        let original_listened_addresses = self.original_listened_addresses.read();
-        self.listened_addresses(max_urls.saturating_sub(original_listened_addresses.len()))
+        let original_listened_addrs = self.original_listened_addrs.read();
+        self.listened_addrs(max_urls.saturating_sub(original_listened_addrs.len()))
             .into_iter()
-            .filter(|(addr, _)| !original_listened_addresses.contains(addr))
+            .filter(|(addr, _)| !original_listened_addrs.contains(addr))
             .chain(
-                original_listened_addresses
+                original_listened_addrs
                     .iter()
                     .map(|addr| (addr.to_owned(), 1)),
```

Comment:
Create a constant for 1.

Reply:
IMO `1` is a proper constant value here. Even we give a constant `ONE`, we should never give it value other than `1`, it is meaningless.